channel/src/main/java/com/synopsys/integration/alert/channel/msteams/MsTeamsChannel.java

```diff
@@ -59,8 +63,11 @@ public MsTeamsChannel(MsTeamsKey msTeamsKey, Gson gson, AuditAccessor auditAcces
 
     @Override
     public void distributeMessage(DistributionEvent event) throws IntegrationException {
-        FieldUtility fields = event.getFieldUtility();
-        String webhook = fields.getString(MsTeamsDescriptor.KEY_WEBHOOK)
+        DistributionJobModel distributionJobModel = event.getDistributionJobModel();
+        DistributionJobDetailsModel distributionJobDetails = distributionJobModel.getDistributionJobDetails();
+        MSTeamsJobDetailsModel asMSTeamsJobDetails = distributionJobDetails.getAsMSTeamsJobDetails();
+        String webhook = Optional.ofNullable(asMSTeamsJobDetails.getWebhook())
```

Comment:
We shouldn't be creating this inline optional object to act as our **if** statement. I believe an if statement here would be much clearer

Reply:
I thought we as a team decided this was the preferred way to handle these cases?